extensions/typescript-language-features/src/languageFeatures/refactor.ts

```diff
@@ -187,36 +187,67 @@ class MoveToFileRefactorCommand implements Command {
 		type DestinationItem = vscode.QuickPickItem & { readonly file: string };
 
 		const workspaceFolder = vscode.workspace.getWorkspaceFolder(document.uri);
-		const destinationItems = response.body.files.map((file): DestinationItem => {
-			const uri = this.client.toResource(file);
-			const parentDir = Utils.dirname(uri);
-
-			let description;
-			if (workspaceFolder) {
-				if (uri.scheme === Schemes.file) {
-					description = path.relative(workspaceFolder.uri.fsPath, parentDir.fsPath);
+
+		const quickPick = vscode.window.createQuickPick();
+		const body = response.body;
+		const updateItems = () => {
+			const destinationItems = body.files.map((file): DestinationItem | undefined => {
+				const uri = this.client.toResource(file);
+				const parentDir = Utils.dirname(uri);
+				const filename = Utils.basename(uri);
+
+				let description;
+				const filterQuery = ['./', '../'].find(str => quickPick.value.startsWith(str));
+				if (workspaceFolder) {
+					if (uri.scheme === Schemes.file) {
+						description = path.relative(workspaceFolder.uri.fsPath, parentDir.fsPath);
+					} else {
+						description = path.posix.relative(workspaceFolder.uri.path, parentDir.path);
+					}
+					if (filterQuery) {
+						const convertRelativePath = (str: string) => {
+							return !str.startsWith('../') ? `./${str}` : str;
+						};
+
+						const relativePath = convertRelativePath(path.relative(path.dirname(document.uri.fsPath), uri.fsPath));
```

Comment:
Can this be updated to support related parent directory accesses, such a `../../`?

Reply:
As I remember it is already possible, but yeah relative paths like ../../dir/file.ts would be already displayed even when just `../` is typed
. I can update the logic to do more strict directory path filtering, but in case if `../` is typed and there is an option for `../dir/file.ts` path, should I also get completions for other directories like `dir` in this case? If so, in this case it would also make sense to file / folder symbol codicon in the start of the label.

Reply:
I will describe it graphically (or attach screenshots) when I'm at home